server/service/src/document/patient/update.rs

```diff
@@ -83,16 +86,30 @@ impl From<RepositoryError> for UpdatePatientError {
     }
 }
 
-fn generate(user_id: &str, patient: &SchemaPatient, input: UpdatePatient) -> RawDocument {
-    RawDocument {
+fn generate(
+    connection: &StorageConnection,
+    user_id: &str,
+    patient: &SchemaPatient,
+    input: UpdatePatient,
+) -> Result<RawDocument, RepositoryError> {
+    let form_schema_id = match input.schema_id {
```

Comment:
Hmm, not quite sure what this change is for, there is something about ` set default Patient schema id in the patient endpoint` in the issue, but not sure what that means. 
Also why would schema_id be optional in input ? I thought we would validate input from F/E against json schema

Reply:
Agree I made the schema_id required. I thought its easier for the FE but FE knows the schema_id anyway so no need to have it optional...

Based on these changes I got a bit sidetracked and added some schema mock data and document schema validation tests.

Reply:
thanks, was wondering, why is schema_id optional on RawDocument and ` #[serde(skip_serializing_if = "Option::is_none")]` ?

Reply:
For the RawDocument its part of the "low level" document api and I don't want to put too many restrictions on it (at least not at this stage), e.g. could use documents without schema if needed.

For the serde macro there is no special reason for it, think I just wanted to make the object smaller if there is no schema...